app/lib/audit_helper.rb

```diff
@@ -1,8 +1,10 @@
 module AuditHelper
   def audit(actor)
-    impersonator = actor.try(:impersonator)
-
-    Audited.audit_class.as_user(impersonator || actor) do
+    if ::Audited.store[:audited_user].blank?
```

Comment:
It's a bit tricky to understand what's going on here without a spec, imo — please could you add one?

Reply:
There was one, in `spec/lib`. I've renamed it accordingly (`ImpersonationAuditHelper`).